src/uct/ib/dc/dc_mlx5.c

```diff
@@ -808,45 +809,79 @@ void uct_dc_mlx5_destroy_dct(uct_dc_mlx5_iface_t *iface)
 }
 #endif
 
-void uct_dc_mlx5_iface_dcis_destroy(uct_dc_mlx5_iface_t *iface, int max)
+static void uct_dc_mlx5_iface_dci_pool_destroy(uct_dc_mlx5_dci_pool_t *dci_pool)
+{
+    ucs_arbiter_cleanup(&dci_pool->arbiter);
+    ucs_free(dci_pool->stack);
+}
+
+static void uct_dc_mlx5_iface_dcis_destroy(uct_dc_mlx5_iface_t *iface,
+                                           uint8_t num_dci_pools,
+                                           uint8_t num_dcis)
```

Comment:
no need for these parameters, they can be taken from `iface`

Reply:
they can't be taken in `uct_dc_mlx5_iface_create_dcis` when we roll back created DCIs